packages/core/assistant/src/session/session.ts

```diff
@@ -154,9 +122,11 @@ export class AiSession {
           Stream.runCollect,
           Effect.map(Chunk.toArray),
         );
+
+        // TODO(burdon): Comment.
```

Comment:
this was necessary to coordinate between the block and message queues so that the streaming Rx value has a signal to clear itself when the message was emitted and not render the message twice.

Dima also has some concerns about consumers coming in partway through a session and there being a backlog in the blocks queue, so prior to this anything unconsumed in the block queue should probably be flushed as well.